blockchain/filedao/staging_buffer.go

```diff
@@ -35,14 +35,10 @@ func (s *stagingBuffer) Get(pos uint64) (*block.Store, error) {
 	return s.buffer[pos], nil
 }
 
-func (s *stagingBuffer) Put(pos uint64, blkBytes []byte) (bool, error) {
+func (s *stagingBuffer) Put(pos uint64, blk *block.Store) (bool, error) {
```

Comment:
directly pass in `*block.Store`, then we can save the extra de-serialization below

Reply:
be careful about pointer sharing

Reply:
after checking, using the incoming `blk` may not be a good idea, since caller may modify it.
and now i recall that's why it is not designed that way in the first place, so close this PR